app/lib/features/events/providers/event_providers.dart

```diff
@@ -1,225 +1,185 @@
-      return eventStartTime.isBefore(currentTime);
-    }).toList();
-    return pastEventList;
-  }
-
-  @override
-  Future<List<ffi.CalendarEvent>> build(String arg) async {
-    final client = ref.watch(alwaysClientProvider);
-    _listener = client.subscribeStream(arg); // keep it resident in memory
-    _listener.forEach((e) async {
-      state = await AsyncValue.guard(() => _getEvents(arg));
-    });
-    final eventsList = await _getEvents(arg);
-    return _sortEventListDscTime(eventsList);
-  }
-}
-
+//SINGLE CALENDER EVENT DETAILS PROVIDER
 final calendarEventProvider = AsyncNotifierProvider.autoDispose
     .family<AsyncCalendarEventNotifier, ffi.CalendarEvent, String>(
   () => AsyncCalendarEventNotifier(),
 );
 
-class AsyncCalendarEventNotifier
-    extends AutoDisposeFamilyAsyncNotifier<ffi.CalendarEvent, String> {
-  late Stream<bool> _listener;
+//MY RSVP STATUS PROVIDER
+final myRsvpStatusProvider = FutureProvider.family
+    .autoDispose<ffi.OptionRsvpStatus, String>((ref, calendarId) async {
+  final event = await ref.watch(calendarEventProvider(calendarId).future);
+  return await event.respondedByMe();
+});
+
+//SpaceId == null : GET LIST OF ALL PINs
+//SpaceId != null : GET LIST OF SPACE PINs
+final allEventListProvider = AsyncNotifierProvider.family<EventListNotifier,
+    List<ffi.CalendarEvent>, String?>(
+  () => EventListNotifier(),
+);
 
-  Future<ffi.CalendarEvent> _getCalendarEvent() async {
-    final client = ref.read(alwaysClientProvider);
-    return await client.waitForCalendarEvent(arg, null);
+//ALL ONGOING EVENTS
+final allOngoingEventListProvider = FutureProvider.autoDispose
+    .family<List<ffi.CalendarEvent>, String?>((ref, spaceId) async {
+  final allEventList = await ref.watch(allEventListProvider(spaceId).future);
+  List<ffi.CalendarEvent> allOngoingEventList = [];
+  for (final event in allEventList) {
+    if (getEventType(event) == EventFilters.ongoing) {
+      allOngoingEventList.add(event);
+    }
   }
+  return sortEventListAscTime(allOngoingEventList);
+});
 
-  @override
-  Future<ffi.CalendarEvent> build(String arg) async {
-    final client = ref.watch(alwaysClientProvider);
-    _listener = client.subscribeStream(arg); // keep it resident in memory
-    _listener.forEach((e) async {
-      state = await AsyncValue.guard(_getCalendarEvent);
-    });
-    return await _getCalendarEvent();
+//MY ONGOING EVENTS
+final myOngoingEventListProvider = FutureProvider.autoDispose
+    .family<List<ffi.CalendarEvent>, String?>((ref, spaceId) async {
+  List<ffi.CalendarEvent> allOngoingEventList =
+      await ref.watch(allOngoingEventListProvider(spaceId).future);
+  List<ffi.CalendarEvent> myOngoingEventList = [];
+  for (final event in allOngoingEventList) {
+    final myRsvpStatus = await ref
+        .watch(myRsvpStatusProvider(event.eventId().toString()).future);
+    if (myRsvpStatus.statusStr() != null && myRsvpStatus.statusStr() == 'yes') {
```

Comment:
```suggestion
    if (myRsvpStatus.statusStr() == 'yes') {
```

means it is `!= null` ...